internal/mode/static/nginx/config/servers.go

```diff
@@ -487,6 +493,50 @@ func generateProxySetHeaders(filters *dataplane.HTTPFilters) []http.Header {
 	return append(proxySetHeaders, headers...)
 }
 
+func generateAddResponseHeaders(filters *dataplane.HTTPFilters) []http.Header {
+	headers := make([]http.Header, len(baseHeaders))
+	copy(headers, baseHeaders)
+
+	if filters == nil || filters.ResponseHeaderModifiers == nil {
+		return headers
+	}
+	headerFilter := filters.ResponseHeaderModifiers
+
+	headerLen := len(headerFilter.Add) + len(headers)
+	responseAddHeaders := make([]http.Header, 0, headerLen)
+	if len(headerFilter.Add) > 0 {
+		// FIXME(kevin85421): Should we use a different function?
+		addHeaders := convertSetHeaders(headerFilter.Add)
+		responseAddHeaders = append(responseAddHeaders, addHeaders...)
+	}
+	return append(responseAddHeaders, headers...)
+}
+
+func generateSetResponseHeaders(filters *dataplane.HTTPFilters) []http.Header {
+	if filters == nil || filters.ResponseHeaderModifiers == nil {
+		return []http.Header{}
+	}
+	headerFilter := filters.ResponseHeaderModifiers
+	headerLen := len(headerFilter.Set)
```

Comment:
nit: no need to declare a variable here, just access the length in the `make` function.

Reply:
Update https://github.com/nginxinc/nginx-gateway-fabric/pull/1494/commits/95e60390f78674d32e283630578bf3cbb922d137.